icaruscode/CRT/CRTUtils/CRTT0MatchAlg.cc

```diff
@@ -793,5 +851,231 @@ namespace icarus{
+     if(fIsData){
+      if (fTSMode == 1) {
+	crtTime = ((double)(int)crtHit.ts1_ns) * 1e-3; //+ fTimeCorrection;
+      }//end if (fTSMode == 1)
+      else {
+		crtTime = double(crtHit.ts0_ns - (trigger_timestamp%1000000000))/1e3;
+		if(crtTime<-0.5e6) 	crtTime+=1e6;
+		else if(crtTime>0.5e6) 	crtTime-=1e6;		
+      }//end else
+     }//end if(fIsData)
+     else if(!fIsData){	
+	crtTime = crtHit.ts0_ns/1e3;
+     }//end else if(!fIsData)
+      // If track is stitched then try all hits
+      if (!((crtTime >= t0MinMax.first - 10. && crtTime <= t0MinMax.second + 10.) 
+            || t0MinMax.first == t0MinMax.second)) continue;
+
+      // cut on CRT hit PE value
+      if (crtHit.peshit<fPEcut) continue;
+      if (crtHit.x_err>fMaxUncert) continue;
+      if (crtHit.y_err>fMaxUncert) continue;
+      if (crtHit.z_err>fMaxUncert) continue;
+      if (tpcTrack.Length() < fMinTrackLength) continue;
+
+      TVector3 crtPoint(crtHit.x_pos, crtHit.y_pos, crtHit.z_pos);
+
+      //Calculate Track direction
+      std::pair<TVector3, TVector3> startEndDir;
+      // dirmethod=2 is original algorithm, dirmethod=1 is simple algorithm for which SCE corrections are possible
+      if (fDirMethod==2)  startEndDir = TrackDirectionAverage(tpcTrack, fTrackDirectionFrac);
+      else startEndDir = TrackDirection(detProp, tpcTrack, fTrackDirectionFrac, crtTime, driftDirection);
+      TVector3 startDir = startEndDir.first;
+      TVector3 endDir = startEndDir.second;
+    
+      // Calculate the distance between the crossing point and the CRT hit, SCE corrections are done inside but dropped
+      double startDist = DistOfClosestApproach(detProp, start, startDir, crtHit, driftDirection, crtTime);
+      double endDist = DistOfClosestApproach(detProp, end, endDir, crtHit, driftDirection, crtTime);
+
+      double xshift = driftDirection * crtTime * detProp.DriftVelocity();
+      auto thisstart = start; 
+      thisstart.SetX(start.X()+xshift);
+      auto thisend = end; 
+      thisend.SetX(end.X()+xshift);
+
+      // repeat SCE correction for endpoints
+      if (fSCE->EnableCalSpatialSCE() && fSCEposCorr) {
+	geo::Point_t temppt = {thisstart.X(),thisstart.Y(),thisstart.Z()};
+	geo::TPCID tpcid = fGeometryService->PositionToTPCID(temppt);
+	geo::Vector_t  fPosOffsets = fSCE->GetCalPosOffsets(temppt,tpcid.TPC);
+	thisstart[0] += fPosOffsets.X();
+	thisstart[1] += fPosOffsets.Y();
+	thisstart[2] += fPosOffsets.Z();
+	temppt.SetX(thisend.X());
+	temppt.SetY(thisend.Y());
+	temppt.SetZ(thisend.Z());
+	tpcid = fGeometryService->PositionToTPCID(temppt);
+	fPosOffsets = fSCE->GetCalPosOffsets(temppt,tpcid.TPC);
+	thisend[0] += fPosOffsets.X();
+	thisend[1] += fPosOffsets.Y();
+	thisend[2] += fPosOffsets.Z();
+      }
+
+      if (startDist<fDistanceLimit || endDist<fDistanceLimit) {
+	double distS = (crtPoint-thisstart).Mag();
+	double distE =  (crtPoint-thisend).Mag();
+	if (distS <= distE && startDist<fDistanceLimit){ 
```

Comment:
As commented above, merging the common branch of these two `if` branches helps maintainability.